capstone/src/main/java/com/google/sps/data/Classroom.java

```diff
@@ -1,94 +1,100 @@
+        this.entity.setProperty(FORMS, new ArrayList<>());
     }
 
-    public void setKey(Key key) {
-        this.key = key;
+    public void setTeacher(User teacher) { 
+        this.entity.setProperty(TEACHER, teacher.getUserKey());
     }
-
-    // Getters
-    public Teacher getTeacher() {
-        return this.teacher;
+    
+    public Key getTeacher() {
+        return (Key) this.entity.getProperty(TEACHER);
     }
 
     public String getSubject() {
-        return this.subject;
+        return (String) this.entity.getProperty(SUBJECT);
     }
 
-    public List<Student> getAllStudents() {
-        return this.students;
+    public Entity getClassroomEntity() {
+        return this.entity;
     }
 
-    public List<Form> getForms() {
-        return this.forms;
+    @SuppressWarnings("unchecked")
+    public List<Key> getAllStudents() {
+        return (ArrayList<Key>) this.entity.getProperty(STUDENTS);
     }
 
-    public Key getKey() {
-        assert this.key.getId() != -1: "The key doesn't exist.";
-        return this.key;
+    @SuppressWarnings("unchecked")
+    public List<Key> getAllForms() { 
+        return (ArrayList<Key>) this.entity.getProperty(FORMS); 
     }
+    
+    @SuppressWarnings("unchecked")
+    public void addStudent(User student) {
+        List<Key> students = (ArrayList<Key>) this.entity.getProperty(STUDENTS);
+        students.add(student.getUserKey());
 
-    // Database
-    public void addStudent(Student student) { 
-        students.add(student);
+        this.entity.setProperty(STUDENTS, students);
     }
 
+    @SuppressWarnings("unchecked")
+    public void removeStudent(User student) {
+        List<Key> students = (ArrayList<Key>) this.entity.getProperty(STUDENTS);
+        students.remove(student.getUserKey());
+
+        this.entity.setProperty(STUDENTS, students);
+    }
+
+    @SuppressWarnings("unchecked")
     public void addForm(Form form) {
-        forms.add(form);
+        List<Key> forms = (ArrayList<Key>) this.entity.getProperty(FORMS);
+        forms.add(form.getFormKey());
+
+        this.entity.setProperty(FORMS, forms);
     }
 
-    public void removeStudent(Student student) {
-        this.students.remove(student);
+    @SuppressWarnings("unchecked")
+    public void removeForm(Form form) {
+        List<Key> forms = (ArrayList<Key>) this.entity.getProperty(FORMS);
+        forms.remove(form.getFormKey());
+
+        this.entity.setProperty(FORMS, forms);
     }
 
-    public boolean isStudentInClass(Student student) {
+    @SuppressWarnings("unchecked")
+    public boolean isStudentInClass(User student) {
+        List<Key> students = (ArrayList<Key>) this.entity.getProperty(STUDENTS);
+
         assert students.size() > 0: "There are no students in this class!";
 
-        if (students.lastIndexOf(student) == -1) {
+        if (students.lastIndexOf(student.getUserKey()) == -1) {
             return false;
         } else {
             return true;
         }
     }
```

Comment:
Whenever you see the pattern:
if (somecondition) {
    return true;
} else {
   return false;
}
you can simplify to 
return somecondition;
(or in your case !somecondition)

Reply:
got it